symphony-bdk-legacy/sms-sdk-renderer-java/src/test/java/it/services/SmsRendererTest.java

```diff
@@ -0,0 +1,160 @@
+
+    if (smsTypes != SmsRenderer.SmsTypes.TABLE) {
+      final String fileName = typeName + "JsonArray.json";
+      jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+      jsonArray = new JSONArray();
+      jsonArray.add(jsonMessageContext);
+    } else {
+      final JSONParser parser = new JSONParser();
+      final String sourceFile = System.getProperty("user.dir") + "/src/test/resources/SmsRenderer/JSONData/tableJsonArray.json";
+      final Object obj = parser.parse(new FileReader(sourceFile));
+      final JSONObject jsonObject = (JSONObject) obj;
+      jsonArray = (JSONArray) jsonObject.get("phones");
+    }
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonArray, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "JsonArrayHtml.html");
+  }
+
+  private void verifyResultJSON(final String typeName, final String compiledTemplate, final String fileName) throws IOException {
+    final String expectedResult = this.minifyHTML(this.getStringFromFile("/SmsRenderer/HTMLResult/" + typeName + fileName));
+    assertNotNull(expectedResult);
+    assertEquals(expectedResult, compiledTemplate);
+  }
+
+  private void testString(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String jsonFileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + jsonFileName);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonMessageContext, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "StringHtml.html");
+  }
+
+  private void testJsonObject(final SmsRenderer.SmsTypes smsTypes) throws IOException, ParseException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String fileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+
+    final JSONObject jsonObject = this.getJsonObject(jsonMessageContext);
+    assertNotNull(jsonObject);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonObject, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "JsonObjectHtml.html");
+  }
+
+  private String getStringFromFile(final String fileName) throws IOException {
+    assertNotNull(fileName);
+
+    final String jsonMessageContext = this.removeCarriageReturn(this.readResourceContent(fileName));
+    assertNotNull(jsonMessageContext);
+    return jsonMessageContext;
+  }
+
+  private String readResourceContent(final String path) throws IOException {
+    assertNotNull(path);
+    final String sourceFile = System.getProperty("user.dir") + "/src/test/resources/" + path;
```

Comment:
`System.getProperty("user.dir")` depends from where you run the process. So, we cannot be sure it will be always the same value. You could get an input stream this way: https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/cfec987f6f1963fb52e02d76ef2335bc9e98c4c1/symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigParserTest.java#L15
`getClass().getResourceAsStream(path)` will give you the correct `InputStream`

Reply:
I don't why when I use the `getClass().getResourceAsStream(path)` it includes the folder `build` in the path result